R/remote.R

```diff
@@ -24,10 +24,29 @@
 #' remote_con(mf2)
 #' remote_query(mf2)
 remote_name <- function(x) {
-  if (!inherits(x$lazy_query, "lazy_query_base"))
+  if (inherits(x$lazy_query, "lazy_query_base")) {
+    return(x$lazy_query$x)
+  }
+
+  if (!inherits(x$lazy_query, "lazy_select_query")) {
+    return()
+  }
+
+  lq <- x$lazy_query
+  if (!inherits(lq$from, "lazy_query_base")) {
+    return()
+  }
+
+  vars_base <- op_vars(lq$from)
+  if (!is_select_trivial(lq$select, vars_base)) {
+    return()
+  }
+
+  if (!(is_empty(lq$where) && is_empty(lq$order_by) && is_false(lq$distinct) && is_empty(lq$limit))) {
     return()
+  }
```

Comment:
Longer, but easier to read?

```suggestion
  if (!is_empty(lq$where)) {
    return()
  }

  if (!is_empty(lq$order_by)) {
    return()
  }

  if (!is_false(lq$distinct)) {
    return()
  }

  if (!is_empty(lq$limit)) {
    return()
  }
```